modules/lang-painless/src/test/java/org/elasticsearch/painless/BaseClassTests.java

```diff
@@ -28,27 +28,27 @@ public class BaseClassTests extends ScriptTestCase {
 
     protected Map<ScriptContext<?>, List<Whitelist>> scriptContexts() {
         Map<ScriptContext<?>, List<Whitelist>> contexts = new HashMap<>();
-        contexts.put(Gets.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(NoArgs.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(OneArg.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ArrayArg.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(PrimitiveArrayArg.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(DefArrayArg.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ManyArgs.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(VarArgs.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(DefaultMethods.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ReturnsVoid.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ReturnsPrimitiveBoolean.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ReturnsPrimitiveInt.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ReturnsPrimitiveFloat.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(ReturnsPrimitiveDouble.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(NoArgsConstant.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(WrongArgsConstant.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(WrongLengthOfArgConstant.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(UnknownArgType.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(UnknownReturnType.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(UnknownArgTypeInArray.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
-        contexts.put(TwoExecuteMethods.CONTEXT, PainlessPlugin.BASE_WHITELISTS);
+        contexts.put(Gets.CONTEXT, PainlessPlugin.baseWhiteList());
```

Comment:
since each call to baseWhitelist now loads the resource, let's put this into a test case static member so we only load it once? it can be reused throughout this suite. In fact perhaps it could still be held onto in ScriptTestCase and referenced in all these test classes?

Reply:
++ Makes sense, added a constant to `ScriptTestCase` :)